### PR TITLE
[Enhance] Add DINO Swin-L

### PR DESCRIPTION
**Pull request by okotaku**
## Motivation

Add DINO Swin-L config + weight.
I uploaded pretrained weight and log here. https://github.com/okotaku/dethub-weights/releases/tag/v0.1.1dino-swinl

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: configs/dino/metafile.yml [hunk 1]
```diff
@@ -48,3 +48,15 @@ Models:
     Results:
       - Task: Object Detection
         Dataset: COCO
+
+  - Name: dino-5scale_swin-l_8xb2-12e_coco.py
```
Comment (RangiLyu) [inline configs/dino/metafile.yml:52]: Great! Would you like to update the README.md? BTW, would you like us to host your model weights on our servers? You can indicate your ownership in the readme and model weight names.

Comment (okotaku) [reply configs/dino/metafile.yml:52]: I would like to host on your servers.

Comment (LYMDLUT) [reply configs/dino/metafile.yml:52]: Would you like to retrain it with convnext v2？
https://github.com/LYMDLUT/mmdetection/blob/3.x/projects/ConvNeXt-V2/configs/dino-4scale_r50_8xb2-12e-fcmae-atto_coco.py

Comment (LYMDLUT) [reply configs/dino/metafile.yml:52]: maybe you can take it as a reference

File: configs/dino/dino-5scale_swin-l_8xb2-12e_coco.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
+_base_ = [
+    '../_base_/datasets/coco_detection.py', '../_base_/default_runtime.py'
```
Comment (Li-Qingyun) [inline configs/dino/dino-5scale_swin-l_8xb2-12e_coco.py:2]: Hi~, thanks for your great work!
I think the config is better to inherit from [dino-4scale_r50_8xb2-12e_coco.py](https://github.com/RistoranteRist/mmdetection/blob/dino-swin/configs/dino/dino-4scale_r50_8xb2-12e_coco.py).

File: configs/dino/dino-5scale_swin-l_8xb2-12e_coco.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+            gamma=2.0,
+            alpha=0.25,
+            loss_weight=1.0),  # 2.0 in DeformDETR
+        loss_bbox=dict(type='L1Loss', loss_weight=5.0),
+        loss_iou=dict(type='GIoULoss', loss_weight=2.0)),
+    dn_cfg=dict(  # TODO: Move to model.train_cfg ?
+        label_noise_scale=0.5,
+        box_noise_scale=1.0,  # 0.4 for DN-DETR
+        group_cfg=dict(dynamic=True, num_groups=None,
+                       num_dn_queries=100)),  # TODO: half num_dn_queries
```
Comment (Li-Qingyun) [inline configs/dino/dino-5scale_swin-l_8xb2-12e_coco.py:92]: Yes, this implementation is referenced from the original dino repository, but the actual meaning is indeed half num_dn_queries. You can modified the name as `num_positive_queries`(but it is also `num_negative_queries`), or modifying `CdnQueryGenerator.get_num_groups()` as `num_dn_queries=200`.

If the confusing name is modified, please help fix other configs such as `dino-4s-r50-12e` and give a comment in the code to remind the users and developers.

Thank you!!

File: configs/dino/dino-5scale_swin-l_8xb2-12e_coco.py [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+        sync_cls_avg_factor=True,
+        loss_cls=dict(
+            type='FocalLoss',
+            use_sigmoid=True,
+            gamma=2.0,
+            alpha=0.25,
+            loss_weight=1.0),  # 2.0 in DeformDETR
+        loss_bbox=dict(type='L1Loss', loss_weight=5.0),
+        loss_iou=dict(type='GIoULoss', loss_weight=2.0)),
+    dn_cfg=dict(  # TODO: Move to model.train_cfg ?
```
Comment (Li-Qingyun) [inline configs/dino/dino-5scale_swin-l_8xb2-12e_coco.py:88]: That's a good idea. @jshilong can provide better suggestions.

And please remember to remove the TODO comments before merging, regardless of whether they are eventually modified

File: configs/dino/metafile.yml (hunk 1 above)
Comment (okotaku) [reply configs/dino/metafile.yml:52]: @LYMDLUT I will try it later.

File: configs/dino/dino-5scale_swin-l_8xb2-12e_coco.py (hunk 3 above)
Comment (okotaku) [reply configs/dino/dino-5scale_swin-l_8xb2-12e_coco.py:92]: The two remaining todos were due to copying comments from dino-4scale_r50_8xb2-12e_coco.py. I will leave this change as a future improvement and not address it here.

File: configs/dino/dino-5scale_swin-l_8xb2-12e_coco.py (hunk 2 above)
Comment (LYMDLUT) [reply configs/dino/dino-5scale_swin-l_8xb2-12e_coco.py]: I cannot agree it more.

File: configs/dino/README.md [hunk 5]
```diff
@@ -14,9 +14,10 @@ We present DINO (DETR with Improved deNoising anchOr boxes), a state-of-the-art
...
+| :------: | :---------: | :-----: | :----: | :---------------------------------------------: | :--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
+|   R-50   | DINO-4scale |   12e   |  49.0  |  [config](./dino-4scale_r50_8xb2-12e_coco.py)   | [model](https://download.openmmlab.com/mmdetection/v3.0/dino/dino-4scale_r50_8xb2-12e_coco/dino-4scale_r50_8xb2-12e_coco_20221202_182705-55b2bba2.pth) \| [log](https://download.openmmlab.com/mmdetection/v3.0/dino/dino-4scale_r50_8xb2-12e_coco/dino-4scale_r50_8xb2-12e_coco_20221202_182705.log.json) |
+|  Swin-L  | DINO-5scale |   12e   |  57.2  | [config](./dino-5scale_swin-l_8xb2-12e_coco.py) |                                                                                                                                          [model](<>) \| [log](<>)                                                                                                                                          |
```
Comment (RangiLyu) [inline configs/dino/README.md:20]: We uploaded the model weight to the cloud.
You can update the link now.
Model: https://download.openmmlab.com/mmdetection/v3.0/dino/dino-5scale_swin-l_8xb2-12e_coco/dino-5scale_swin-l_8xb2-12e_coco_20230228_072924-a654145f.pth
Log: https://download.openmmlab.com/mmdetection/v3.0/dino/dino-5scale_swin-l_8xb2-12e_coco/dino-5scale_swin-l_8xb2-12e_coco_20230228_072924.log

**Review by RangiLyu — APPROVED** [review]

**Pull request merged**